internal/testcli/executor.go

```diff
@@ -361,3 +361,17 @@ func DeployContract(t *testing.T, e *Executor, inPath, configPath, wallet, addre
 	require.NoError(t, err)
 	return h
 }
+
+// CheckLineTxPersisted checks that the next line contains a transaction hash.
+func (e *Executor) CheckLineTxPersisted(t *testing.T) (*transaction.Transaction, uint32) {
```

Comment:
CheckAwaitableTxPersisted

Reply:
It's not different from CheckTxPersisted. Reuse CheckTxPersisted firstly, and afterwards add a code about `OnChain` and etc. lines checks. Take CheckTxTestInvokeOutput as a base.